app/src/main/java/pm/gnosis/heimdall/data/repositories/impls/WalletConnectBridgeRepository.kt

```diff
@@ -156,11 +179,12 @@ class WalletConnectBridgeRepository @Inject constructor(
                 override fun sessionApproved() {}
 
                 override fun sessionClosed() {
-                    sessions.remove(config.handshakeTopic)
+                    sessionsPreferences.edit { remove(KEY_PREFIX_SESSION_SAFE + sessionId) }
```

Comment:
I think it would be better to encapsulate SharedPreferences like this to classes and use getters & setters for editing. Also all respective keys can be kept inside companion object of those classes

Reply:
I think this would be too much for this PR to change. In general a better abstraction from shared preferences would make sense, but I would recommend having this as a separate issue.

Reply:
agree. https://github.com/gnosis/safe-android/issues/399